internal/factsengine/gatherers/sysctl_test.go

```diff
@@ -0,0 +1,184 @@
+	suite.ElementsMatch(expectedResults, factResults)
+}
+
+func (suite *SysctlTestSuite) TestSysctlGathererNonExistingKey() {
+	mockOutputFile, _ := os.Open(helpers.GetFixturePath("gatherers/sysctl.output"))
+	mockOutput, _ := io.ReadAll(mockOutputFile)
+	suite.mockExecutor.On("Exec", "sysctl", "-a").Return(mockOutput, nil)
+
+	c := gatherers.NewSysctlGatherer(suite.mockExecutor)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "madeup_fact",
+			Gatherer: "sysctl",
+			Argument: "madeup.fact",
+		},
+	}
+
+	factResults, err := c.Gather(factRequests)
+
+	expectedResults := []entities.Fact{
+		{
+			Name:  "madeup_fact",
+			Value: nil,
+			Error: &entities.FactGatheringError{
+				Message: "requested value not found in sysctl output: madeup.fact",
+				Type:    "sysctl-value-not-found",
+			},
+		},
+	}
+
+	suite.NoError(err)
+	suite.ElementsMatch(expectedResults, factResults)
+}
+
+func (suite *SysctlTestSuite) TestSysctlCommandNotFound() {
+	suite.mockExecutor.On("Exec", "sysctl", "-a").Return(nil, exec.ErrNotFound)
+
+	c := gatherers.NewSysctlGatherer(suite.mockExecutor)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "fs.inotify.max_user_watches",
+			Gatherer: "sysctl",
+			Argument: "fs.inotify.max_user_watches",
+		},
+	}
+
+	expectedError := &entities.FactGatheringError{
+		Message: "error executing sysctl command: executable file not found in $PATH",
+		Type:    "sysctl-cmd-error",
+	}
+
+	factResults, err := c.Gather(factRequests)
+
+	suite.EqualError(err, expectedError.Error())
+
+	suite.Empty(factResults)
+}
+
+func (suite *SysctlTestSuite) TestSysctlGatherer() {
+	mockOutputFile, _ := os.Open(helpers.GetFixturePath("gatherers/sysctl.output"))
+	mockOutput, _ := io.ReadAll(mockOutputFile)
+	suite.mockExecutor.On("Exec", "sysctl", "-a").Return(mockOutput, nil)
+
+	c := gatherers.NewSysctlGatherer(suite.mockExecutor)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "simple_value",
+			Gatherer: "sysctl",
+			Argument: "fs.inotify.max_user_watches",
+		},
+	}
+
+	factResults, err := c.Gather(factRequests)
+
+	expectedResults := []entities.Fact{
+		{
+			Name:  "simple_value",
+			Value: &entities.FactValueInt{Value: 65536},
+		},
+	}
+
+	suite.NoError(err)
+	suite.ElementsMatch(expectedResults, factResults)
+}
+
+func (suite *SysctlTestSuite) TestSysctlGathererEmptyValue() {
+	mockOutputFile, _ := os.Open(helpers.GetFixturePath("gatherers/sysctl.output"))
+	mockOutput, _ := io.ReadAll(mockOutputFile)
+	suite.mockExecutor.On("Exec", "sysctl", "-a").Return(mockOutput, nil)
+
+	c := gatherers.NewSysctlGatherer(suite.mockExecutor)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "empty_value",
+			Gatherer: "sysctl",
+			Argument: "kernel.domainname",
+		},
+	}
+
+	factResults, err := c.Gather(factRequests)
+
+	expectedResults := []entities.Fact{
+		{
+			Name:  "empty_value",
+			Value: &entities.FactValueString{Value: ""}, // We should probably use the upcoming FactValueNil (PR#263)
```

Comment:
Up to you. No hard opinions.
Remember to document the way you use in both cases